File: c++/1015.cpp

#include <iostream>
#include <cstdio>
#include <cmath>
#include <vector>
#include <algorithm>
#include <queue>
#include <set>
#include <string>
#include <cstring>
#include <map>
#include <stack>

using namespace std;



int change(int a,int b){
    int arr[1000];
    int index = 0;
    while (a) {
        int d = a % b;
        arr[index++] = d;
        a = a / b;
    }
    int base = 0;
    int sum = 0;
    for (int i=index-1; i>=0; i--) {
        sum = arr[i]*pow(b, base) + sum;
        base++;
    }
    return sum;
}



bool isPrimer(int x){
    if (x <= 1) {
        return false;
    }
    for (int i=2; i<=sqrt(x); i++) {
        if (x % i == 0) {
            return false;
        }
    }
    return true;
}
int main(){
    int n,radix;
    while (scanf("%d",&n)!=EOF) {
        
        if (n < 0) {
            break;
        }
        scanf("%d",&radix);
        if (isPrimer(n)) {
            int a = change(n, radix);
            if (isPrimer(a)) {
                printf("Yes\n");
            }else
                printf("No\n");
        }else
            printf("No\n");
        
    }
    return 0;
}

File: Python3/1015.py
from math import sqrt
def isPrimer(n):
    if n <= 1:
        return False
    for i in range(2,int(sqrt(n))+1):
        if n % i == 0:
            return False
    return True
def convert(n,radix):
    data =[]
    while n!=0:
        t = n % radix
        n = n//radix
        data.append(t)
    ans = sum(j*pow(radix,i) for i,j in enumerate(data[::-1]))
    return  ans
while True:
    a = list(map(int,input().split()))
    if a[0] < 0:
        break
    if isPrimer(a[0]):
        if isPrimer(convert(a[0],a[1])):
            print("Yes")
        else:
            print("No")
    else:
        print("No")
